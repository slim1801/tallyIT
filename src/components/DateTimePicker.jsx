import React, { Component } from "react";

import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const styles = theme => ({
  root: {
    width: "100%"
  },
  textField: {
    color: "white"
  },
  underline: {
    "&:before": {
      borderBottom: "1px solid white"
    },
    "&:hover:not(disabled):not(focused):not(errored):before": {
      borderColor: "white !important"
    },
    "&:after": {
      borderBottom: "2px solid white"
    }
  }
});

export function DateTimePicker(props) {
  const { classes, value } = props;

  const onDateTimeChange = event => {
    const { value } = event.currentTarget;
    props.onDateTimeChanged(value);
  };

  return (
    <form noValidate>
      <TextField
        label={props.label}
        classes={{ root: classes.root }}
        type="datetime-local"
        InputProps={{
          classes: {
            input: classes.textField,
            underline: classes.underline
          }
        }}
        value={value}
        onChange={onDateTimeChange}
      />
    </form>
  );
}

export default withStyles(styles)(DateTimePicker);
