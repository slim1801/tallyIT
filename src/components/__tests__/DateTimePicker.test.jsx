import React from "react";
import DateTimePicker from "../DateTimePicker";

describe("DateTimePicker", () => {
  test("smoke test", () => {
    const DateTimePickerWrapper = shallow(<DateTimePicker />);
    expect(DateTimePickerWrapper).toMatchSnapshot();
  });
});
