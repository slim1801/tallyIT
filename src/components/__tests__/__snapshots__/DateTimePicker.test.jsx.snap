// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker smoke test 1`] = `ShallowWrapper {}`;
